### PR TITLE
fix: make scene `save as` visible

### LINKED ISSUE #8656: [BUG] "save as" is broken in the Studio
Save As UI is broken.  No button to submit and proportions don't match the input fields 

Image

### PR DESCRIPTION
**Pull request by aditya-mitra**
## Summary

- save scene as png
- make the component fields visible
- change pixel ratio before resizing window

image

## References

closes #8656 

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/editor/src/functions/takeScreenshot.ts [hunk 1]
```diff
@@ -200,29 +200,29 @@ export async function takeScreenshot(
...
+    EngineRenderer.instance.effectComposer.setSize(width, height, false)
+    EngineRenderer.instance.renderer.setPixelRatio(1)
+
     const interval = setInterval(() => {
       const viewport = EngineRenderer.instance.renderContext.getParameter(
         EngineRenderer.instance.renderContext.VIEWPORT
       )
-      // todo - scrolling in and out sometimes causes weird pixel ratios that can cause this to fail
-      if (viewport[2] === Math.round(width * pixelRatio) && viewport[3] === Math.round(height * pixelRatio)) {
+      if (viewport[2] === Math.round(width) && viewport[3] === Math.round(height)) {
```
Comment (HexaField) [inline packages/editor/src/functions/takeScreenshot.ts:212]: Pixel ratio must be preserved here

File: packages/editor/src/functions/takeScreenshot.ts [hunk 2]
```diff
@@ -200,29 +200,29 @@ export async function takeScreenshot(
...
     }, 10)
+
     const timeout = setTimeout(() => {
       console.warn('Could not resize viewport in time')
       clearTimeout(timeout)
       clearInterval(interval)
       reject()
     }, 10000)
-
-    // set up effect composer
```
Comment (HexaField) [inline packages/editor/src/functions/takeScreenshot.ts:222]: Why was this moved earlier up?

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/editor/src/functions/takeScreenshot.ts (hunk 2 above)
Comment (aditya-mitra) [reply packages/editor/src/functions/takeScreenshot.ts:222]: After testing this a lot of times, the timer functions are called before the resizing of the scene. So, the `if` condition above does not ever get satisfied and an error is thrown.
Synchronously calling them does not cause that error.

File: packages/editor/src/functions/takeScreenshot.ts (hunk 1 above)
Comment (aditya-mitra) [reply packages/editor/src/functions/takeScreenshot.ts:212]: We are changing the pixel ratio to 1.
Also tried this a few times, `viewport` comes out to be like `0 0 512 320` and `height` and `width` are also `320` and `512` respectively but `pixelRatio` is `2`. So, the condition is not fulfilled. Also,`pixelRatio` is defined outside the promise so it does not change even though the promise inside changes the pixelRatio to 1.

**Review by HexaField — APPROVED** [review]

**Pull request merged**